src/document/Document.js

```diff
@@ -698,7 +698,24 @@ define(function (require, exports, module) {
     Document.prototype.isUntitled = function () {
         return this.file instanceof InMemoryFile;
     };
+
+    /**
+     * Set lineEnding of the document. Change Text in the process.
+     */
+    Document.prototype.setLineEndings = function (lineEndings) {
+        this._lineEndings = lineEndings;
+        this.setText(FileUtils.translateLineEndings(this.getText(true), lineEndings));
```

Comment:
You don't need this line -- the document model's text uses normalized line endings, so this is a no-ope anyway.  Removing this line should fix the cursor/scroll position issues too.